stage1/usr_from_kvm/lkvm.mk

```diff
@@ -5,39 +5,55 @@ LKVM_BINARY := $(LKVM_SRCDIR)/lkvm-static
 LKVM_ACI_BINARY := $(ACIROOTFSDIR)/lkvm
 LKVM_GIT := https://kernel.googlesource.com/pub/scm/linux/kernel/git/will/kvmtool
 # just last published version (for reproducible builds), not for any other reason 
-# cat be set to master to follow to upstream
-# LKVM_VERSION := master
 LKVM_VERSION := 4095fac878f618ae5e7384a1dc65ee34b6e05217
 
+LKVM_STUFFDIR := $(MK_SRCDIR)/lkvm
+LKVM_PATCHES := $(abspath $(LKVM_STUFFDIR)/patches/*.patch)
+
+$(call setup-stamp-file,LKVM_PATCH_STAMP,/patch_lkvm)
+$(call setup-dep-file,LKVM_PATCHES_DEPMK)
+
 UFK_STAMPS += $(LKVM_STAMP)
 INSTALL_FILES += $(LKVM_BINARY):$(LKVM_ACI_BINARY):-
 CREATE_DIRS += $(LKVM_TMP)
 
+
 $(LKVM_STAMP): $(LKVM_ACI_BINARY)
 	touch "$@"
 
 $(LKVM_BINARY): LKVM_SRCDIR := $(LKVM_SRCDIR)
-$(LKVM_BINARY):
+$(LKVM_BINARY): $(LKVM_PATCH_STAMP)
 	$(MAKE) -C "$(LKVM_SRCDIR)" lkvm-static
 
+-include $(LKVM_PATCHES_DEPMK)
+$(LKVM_PATCH_STAMP): GO_ENV := $(GO_ENV)
+$(LKVM_PATCH_STAMP): DEPSGENTOOL := $(DEPSGENTOOL)
+$(LKVM_PATCH_STAMP): LKVM_PATCHES := $(LKVM_PATCHES)
+$(LKVM_PATCH_STAMP): LKVM_PATCHES_DEPMK := $(LKVM_PATCHES_DEPMK)
+$(LKVM_PATCH_STAMP): LKVM_SRCDIR := $(LKVM_SRCDIR)
+$(LKVM_PATCH_STAMP): $(LKVM_SRCDIR)/Makefile $(DEPSGENTOOL_STAMP)
+	set -e; \
+	shopt -s nullglob; \
+	$(GO_ENV) "$(DEPSGENTOOL)" glob --target '$$(LKVM_SRCDIR)/Makefile' --suffix=.patch $(LKVM_PATCHES) >"$(LKVM_PATCHES_DEPMK)"; \
+	for p in $(LKVM_PATCHES); do \
+		patch --directory="$(LKVM_SRCDIR)" --strip=1 --forward <"$${p}"; \
+	done; \
+	touch "$@"
+
+# add remote only if not added 
+# don't fetch existing (commit cannot change)
 $(LKVM_SRCDIR)/Makefile: LKVM_GIT := $(LKVM_GIT)
 $(LKVM_SRCDIR)/Makefile: LKVM_SRCDIR := $(LKVM_SRCDIR)
 $(LKVM_SRCDIR)/Makefile: LKVM_VERSION := $(LKVM_VERSION)
 $(LKVM_SRCDIR)/Makefile: | $(LKVM_TMP)
 	set -e; \
 	mkdir -p $(LKVM_SRCDIR); cd $(LKVM_SRCDIR); \
-	git init; \
-	git remote add origin "$(LKVM_GIT)"; \
-	git fetch --depth=1 origin $(LKVM_VERSION); \
-	git checkout $(LKVM_VERSION)
-	# git clone --depth=1 "$(LKVM_GIT)" "$(LKVM_SRCDIR)"
-
-GR_TARGET := $(LKVM_BINARY)
-GR_SRCDIR := $(LKVM_SRCDIR)
-GR_BRANCH := master
-GR_PREREQS := $(LKVM_SRCDIR)/Makefile
-
-include makelib/git-refresh.mk
+ 	git init; \
+	git remote | grep --silent origin || git remote add origin "$(LKVM_GIT)"; \
+	git rev-parse --silent --verify HEAD 2>/dev/null || git fetch --depth=1 origin $(LKVM_VERSION) && git checkout -q $(LKVM_VERSION); \
```

Comment:
- `git rev-parse` does not have `--silent` flag, only `--quiet` (at least in the version I have - 2.1.0).
- `--quiet` causes no output to stderr, so it only prints to stdout, so please use `>/dev/null` instead of `2>/dev/null'.
- Please use longer `--quiet` flag instead of `-q` in `git checkout` in that line too.